File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms2/Vec4Uniform.java
package org.saar.lwjgl.opengl.shaders.uniforms2;

import org.joml.Vector4fc;
import org.lwjgl.opengl.GL20;

public abstract class Vec4Uniform extends UniformBase implements Uniform {

    @Override
    public final void doLoad() {
        final Vector4fc v = getUniformValue();
        GL20.glUniform4f(getLocation(), v.x(), v.y(), v.z(), v.w());
    }

    public abstract Vector4fc getUniformValue();
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shader/uniforms/Vec4Uniform.kt
package org.saar.lwjgl.opengl.shader.uniforms

import org.joml.Vector4fc
import org.lwjgl.opengl.GL20

abstract class Vec4Uniform : Uniform {

    final override fun load(location: Int) {
        val value: Vector4fc = this.value
        GL20.glUniform4f(location, value.x(), value.y(), value.z(), value.w())
    }

    abstract val value: Vector4fc
}